inc/builder/type/header/mode-switcher/dynamic-css/dynamic.css.php

```diff
@@ -15,6 +15,51 @@
  */
 add_filter( 'astra_dynamic_theme_css', 'astra_mode_switcher_dynamic_css' );
 
+/**
+ * Generate dark palette CSS variable styles for the front end.
+ *
+ * @since x.x.x
+ * @return string
+ */
+function astra_generate_dark_palette_style() {
+
+	$variable_prefix    = Astra_Global_Palette::get_css_variable_prefix();
+	$dark_palette       = astra_get_option( 'dark-mode-palette', 'palette_2' );
+	$ast_palette_config = astra_get_palette_colors();
+	$palette_style      = array();
+	$palette_css_vars   = array();
+	$css                = '';
+
+	if ( isset( $ast_palette_config['palettes'][ $dark_palette ] ) ) {
+		foreach ( $ast_palette_config['palettes'][ $dark_palette ] as $key => $color ) {
+			$palette_key = str_replace( '--', '-', $variable_prefix ) . $key;
+
+			$palette_style[ 'html.ast-dark-site .has' . $palette_key . '-color' ] = array(
+				'color' => 'var(' . $variable_prefix . $key . ')',
+			);
+
+			$palette_style[ 'html.ast-dark-site .has' . $palette_key . '-background-color' ] = array(
+				'background-color' => 'var(' . $variable_prefix . $key . ')',
+			);
+
+			$palette_style[ 'html.ast-dark-site .wp-block-button .has' . $palette_key . '-color' ] = array(
+				'color' => 'var(' . $variable_prefix . $key . ')',
+			);
+
+			$palette_style[ 'html.ast-dark-site .wp-block-button .has' . $palette_key . '-background-color' ] = array(
+				'background-color' => 'var(' . $variable_prefix . $key . ')',
+			);
```

Comment:
```suggestion
			$palette_style[ '.ast-dark-site .has' . $palette_key . '-color' ] = array(
				'color' => 'var(' . $variable_prefix . $key . ')',
			);

			$palette_style[ '.ast-dark-site .has' . $palette_key . '-background-color' ] = array(
				'background-color' => 'var(' . $variable_prefix . $key . ')',
			);

			$palette_style[ '.ast-dark-site .wp-block-button .has' . $palette_key . '-color' ] = array(
				'color' => 'var(' . $variable_prefix . $key . ')',
			);

			$palette_style[ 'html.ast-dark-site .wp-block-button .has' . $palette_key . '-background-color' ] = array(
				'background-color' => 'var(' . $variable_prefix . $key . ')',
			);
```
@imnavanath I think an additional html selector can be removed.

Reply:
Thanks nice suggestion, Done 👍